### PR TITLE
Issue #4588: Split and Organize Checkstyle inputs for MainTests to 'main' folder

### PR DESCRIPTION
**Pull request by Kietzmann**
#4588 
Files that are left in `/checkstyle` directory is used in the other tests. All dependent resource files that was in `/checkstyle` dir were moved to `/main` directory.

### CODE REVIEW

File: src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java [hunk 1]
```diff
@@ -113,11 +113,15 @@
             new String[] {String.valueOf(1)}, null, getClass(), null);
 
     private static String getPath(String filename) {
-        return "src/test/resources/com/puppycrawl/tools/checkstyle/" + filename;
+        return "src/test/resources/com/puppycrawl/tools/checkstyle/main/" + filename;
+    }
+
+    private static String getNonPackagePath(String filePath) {
+        return "src/test/resources/com/puppycrawl/tools/checkstyle/" + filePath;
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: This type of method should be avoided.

File: src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java [hunk 2]
```diff
@@ -629,21 +633,24 @@ public void testPrintTreeOnMoreThanOneFile() throws Exception {
             assertEquals("", systemErr.getLog());
         });
 
-        Main.main("-t", getPath("checks/metrics"));
+        Main.main("-t", getNonPackagePath(
+                "checks/metrics"));
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: We should be able to use input directory for this as there is more than 1 file in it.

File: src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java [hunk 3]
```diff
@@ -708,27 +717,28 @@ public void testPrintTreeCommentsOption() throws Exception {
     @Test
     public void testPrintTreeJavadocOption() throws Exception {
         final String expected = new String(Files.readAllBytes(Paths.get(
-            getPath("astprinter/expectedInputJavadocComment.txt"))), StandardCharsets.UTF_8)
+            getNonPackagePath("astprinter/expectedInputJavadocComment.txt"))),
+                StandardCharsets.UTF_8)
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: input and expected output should be copied to test's input folder.

File: src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java [hunk 4]
```diff
@@ -708,27 +717,28 @@ public void testPrintTreeCommentsOption() throws Exception {
...
             .replaceAll("\\\\r\\\\n", "\\\\n");
 
         exit.checkAssertionAfterwards(() -> {
             assertEquals(expected, systemOut.getLog().replaceAll("\\\\r\\\\n", "\\\\n"));
             assertEquals("", systemErr.getLog());
         });
-        Main.main("-j", getPath("astprinter/InputJavadocComment.javadoc"));
+        Main.main("-j", getNonPackagePath("astprinter/InputJavadocComment.javadoc"));
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: same.

File: src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java [hunk 5]
```diff
@@ -708,27 +717,28 @@ public void testPrintTreeCommentsOption() throws Exception {
...
     }
 
     @Test
     public void testPrintFullTreeOption() throws Exception {
         final String expected = new String(Files.readAllBytes(Paths.get(
-            getPath("astprinter/expectedInputAstTreeStringPrinterJavadoc.txt"))),
+                getNonPackagePath("astprinter/expectedInputAstTreeStringPrinterJavadoc.txt"))),
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: same.

File: src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java [hunk 6]
```diff
@@ -708,27 +717,28 @@ public void testPrintTreeCommentsOption() throws Exception {
...
             StandardCharsets.UTF_8).replaceAll("\\\\r\\\\n", "\\\\n");
 
         exit.checkAssertionAfterwards(() -> {
             assertEquals(expected, systemOut.getLog().replaceAll("\\\\r\\\\n", "\\\\n"));
             assertEquals("", systemErr.getLog());
         });
-        Main.main("-J", getPath("astprinter/InputAstTreeStringPrinterJavadoc.java"));
+        Main.main("-J", getNonPackagePath("astprinter/InputAstTreeStringPrinterJavadoc.java"));
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: same.

File: src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java [hunk 7]
```diff
@@ -741,7 +751,7 @@ public void testConflictingOptionsTvsC() throws Exception {
             assertEquals("", systemErr.getLog());
         });
 
-        Main.main("-c", "/google_checks.xml", "-t", getPath("checks/metrics"));
+        Main.main("-c", "/google_checks.xml", "-t", getNonPackagePath("checks/metrics"));
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: same as `checks/metrics`

File: src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java [hunk 8]
```diff
@@ -754,7 +764,8 @@ public void testConflictingOptionsTvsP() throws Exception {
             assertEquals("", systemErr.getLog());
         });
 
-        Main.main("-p", getPath("mycheckstyle.properties"), "-t", getPath("checks/metrics"));
+        Main.main("-p", getPath("mycheckstyle.properties"), "-t",
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: `mycheckstyle.properties` should be name after test and start with `Input`.

File: src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java [hunk 9]
```diff
@@ -754,7 +764,8 @@ public void testConflictingOptionsTvsP() throws Exception {
...
+                getNonPackagePath("checks/metrics"));
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: Same as `checks/metrics`.

File: src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java [hunk 10]
```diff
@@ -767,7 +778,7 @@ public void testConflictingOptionsTvsF() throws Exception {
             assertEquals("", systemErr.getLog());
         });
 
-        Main.main("-f", "plain", "-t", getPath("checks/metrics"));
+        Main.main("-f", "plain", "-t", getNonPackagePath("checks/metrics"));
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: Same as `checks/metrics`.

File: src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java [hunk 11]
```diff
@@ -781,7 +792,7 @@ public void testConflictingOptionsTvsO() throws Exception {
             assertEquals("", systemErr.getLog());
         });
 
-        Main.main("-o", file.getCanonicalPath(), "-t", getPath("checks/metrics"));
+        Main.main("-o", file.getCanonicalPath(), "-t", getNonPackagePath("checks/metrics"));
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: Same as `checks/metrics`.

File: src/test/resources/com/puppycrawl/tools/checkstyle/main/config-custom-root-module.xml [hunk 12]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
```
Comment (rnveach) [inline src/test/resources/com/puppycrawl/tools/checkstyle/main/config-custom-root-module.xml]: In latest master, only `MainTest` uses this file, so it should be moved.
Please rebase on latest master if you aren't already.

File: src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java (hunk 1 above)
Comment (Kietzmann) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: Ok
How should I get the path of packages that are in outer scope?

Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: @Kietzmann Every comment I made below discusses that. These files shouldn't be in outer scope but be in MainTest's input folder.

Comment (Kietzmann) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: I mentioned package, not file
` Main.main("-c", "/google_checks.xml", "-t", getPath("checks/metrics"));`
how should I get path with modified version of `getPath` method?

Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: So did I. For `checks/metrics` see https://github.com/checkstyle/checkstyle/pull/4697#discussion_r126657978.
We should use MainTest's input folder, not someone elses.

File: src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java (hunk 2 above)
Comment (Kietzmann) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: I did not understood. I should provide new method or cope files from those directory?

Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: `getNonPackagePath("checks/metrics"));` => `getPath("")`
We should use input directory, `main`, just like we require all input files in that directory.

**Comment by Kietzmann** [comment]
@rnveach 
Seems that I fixed all your comments
Could you please review it?
Thanks

File: config/checkstyle_non_main_files_suppressions.xml [hunk 13]
```diff
@@ -18,6 +18,7 @@
     <!-- Javadoc files should not have new line by design (on purpose). -->
      <suppress checks="NewlineAtEndOfFile" files="[\\/]test[\\/].*[\\/]grammars[\\/]javadoc[\\/].*\.javadoc"/>
      <suppress checks="NewlineAtEndOfFile" files="[\\/]test[\\/].*[\\/]astprinter[\\/].*\.javadoc"/>
+     <suppress checks="NewlineAtEndOfFile" files="[\\/]test[\\/].*[\\/]main[\\/].*\.javadoc"/>
```
Comment (rnveach) [inline config/checkstyle_non_main_files_suppressions.xml]: @romani Should we just suppress all javadoc files?
IE: `files="[\\/]test[\\/].*\.javadoc"`

Comment (romani) [inline config/checkstyle_non_main_files_suppressions.xml]: yes, could be done separately -  #4702

**Comment by Kietzmann** [comment]
@rnveach 
Fixed

**Pull request merged**